x/epochstorage/keeper/migrations.go

```diff
@@ -116,64 +119,146 @@ func (m Migrator) Migrate3to4(ctx sdk.Context) error {
+	// defer iterator.Close()
 
-	for ; iterator.Valid(); iterator.Next() {
-		var stakeStorageV6 v6.StakeStorage
-		m.keeper.cdc.MustUnmarshal(iterator.Value(), &stakeStorageV6)
+	// for ; iterator.Valid(); iterator.Next() {
+	// 	var stakeStorageV6 v6.StakeStorage
+	// 	m.keeper.cdc.MustUnmarshal(iterator.Value(), &stakeStorageV6)
 
-		for i := range stakeStorageV6.StakeEntries {
-			stakeStorageV6.StakeEntries[i].Vault = stakeStorageV6.StakeEntries[i].Address
-		}
+	// 	for i := range stakeStorageV6.StakeEntries {
+	// 		stakeStorageV6.StakeEntries[i].Vault = stakeStorageV6.StakeEntries[i].Address
+	// 	}
 
-		store.Set(iterator.Key(), m.keeper.cdc.MustMarshal(&stakeStorageV6))
-	}
+	// 	store.Set(iterator.Key(), m.keeper.cdc.MustMarshal(&stakeStorageV6))
+	// }
 
 	return nil
 }
 
 // Migrate6to7 goes over all existing stake entries and populates the new description field with current moniker
 func (m Migrator) Migrate6to7(ctx sdk.Context) error {
-	utils.LavaFormatDebug("migrate: epochstorage to include detailed description")
+	// utils.LavaFormatDebug("migrate: epochstorage to include detailed description")
 
-	store := prefix.NewStore(ctx.KVStore(m.keeper.storeKey), types.KeyPrefix(types.StakeStorageKeyPrefix))
+	// store := prefix.NewStore(ctx.KVStore(m.keeper.storeKey), types.KeyPrefix(types.StakeEntriesPrefix))
+	// iterator := sdk.KVStorePrefixIterator(store, []byte{})
+
+	// defer iterator.Close()
+
+	// for ; iterator.Valid(); iterator.Next() {
+	// 	var stakeStorageV7 types.StakeStorage
+	// 	m.keeper.cdc.MustUnmarshal(iterator.Value(), &stakeStorageV7)
+
+	// 	for i := range stakeStorageV7.StakeEntries {
+	// 		stakeStorageV7.StakeEntries[i].Description.Moniker = stakeStorageV7.StakeEntries[i].Moniker
+	// 		stakeStorageV7.StakeEntries[i].Moniker = ""
+	// 	}
+
+	// 	store.Set(iterator.Key(), m.keeper.cdc.MustMarshal(&stakeStorageV7))
+	// }
+
+	return nil
+}
+
+// Migrate7to8 transfers all the stake entries from the old stake storage to the new stake entries store
+// StakeStorage is set to the stake entries store
+// StakeStorageCurrent is set to the stake entries current store
+// StakeStorageUnstake is deleted
+func (m Migrator) Migrate7to8(ctx sdk.Context) error {
+	utils.LavaFormatDebug("migrate: epochstorage to move stake entries from stake storage")
+	k := m.keeper
+
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), v3.KeyPrefix(v3.StakeStorageKeyPrefix))
 	iterator := sdk.KVStorePrefixIterator(store, []byte{})
 
 	defer iterator.Close()
 
 	for ; iterator.Valid(); iterator.Next() {
-		var stakeStorageV7 types.StakeStorage
-		m.keeper.cdc.MustUnmarshal(iterator.Value(), &stakeStorageV7)
+		key := string(iterator.Key())
 
-		for i := range stakeStorageV7.StakeEntries {
-			stakeStorageV7.StakeEntries[i].Description.Moniker = stakeStorageV7.StakeEntries[i].Moniker
-			stakeStorageV7.StakeEntries[i].Moniker = ""
+		// identify stake storage type: regular, current or unstake
+		if key == v3.StakeStorageKeyUnstakeConst {
```

Comment:
manually test the migration worked
- providers still exist
- pairing list for a consumer stays the same for a specific block before and after the change

Reply:
done. fixed migrator here: https://github.com/lavanet/lava/commit/fef9827dc95c35bedb5d7485e337c39012d2ca53, https://github.com/lavanet/lava/commit/1c8f342a969f885caee01cad8ff1fbf7966992fb